### PR TITLE
Feature/crs

### LINKED ISSUE #17: US - save data in a given Crs
As **editor**, I want to **add** or **modify** data in another crs so as not to have to do the conversions on my post

### LINKED ISSUE #18: US - retrieve data in a specific Crs
As **editor**, I want to **read** data in another crs so as not to have to do the conversions on my desktop

### PR DESCRIPTION
**Pull request by azarz**
closes #18 (already implemented, added tests and better error handling)
closes #17

### CODE REVIEW

File: internal/data/catalog_db.go [hunk 1]
```diff
@@ -317,7 +317,11 @@ func (cat *catalogDB) AddTableFeature(ctx context.Context, tableName string, jso
 
 	i++
 	columnStr += ", " + tbl.GeometryColumn
-	placementStr += fmt.Sprintf(", ST_GeomFromGeoJSON($%d)", i)
+	geomStr := fmt.Sprintf(", ST_GeomFromGeoJSON($%d)", i)
+	if crs != "" {
+		geomStr = fmt.Sprintf(", ST_Transform(ST_SetSRID(ST_GeomFromGeoJSON($%d), %s), 4326)", i, crs)
```
Comment (benoitdm-oslandia) [inline internal/data/catalog_db.go]: does it mean you only allow 4326 data in database?

Comment (azarz) [reply internal/data/catalog_db.go]: My bad, fixed in 0da771b.
We now use tbl.Srid (which I didn't know about before)

**Review by benoitdm-oslandia — CHANGES_REQUESTED** [review]
LGTM! Just if you can add a test with projection name like with EPSG:2154 or IGNF:LAMB93... this should do the same job!

**Comment by azarz** [comment]
> LGTM! Just if you can add a test with projection name like with EPSG:2154 or IGNF:LAMB93... this should do the same job!

This would imply a change in the current API, which currently needs a number in the request, corresponding to the SRID in postGIS. An other issue is that the IGNF: SRSs are not available in postGIS...
I tried to find a standard definition of the IGNF:LAMB93 as in http://www.opengis.net/def/crs/EPSG/0/2154, IGNF is not even listed in http://www.opengis.net/def/crs/.

One way to tackle this issue would be to accept `proj` strings, but I don't know if this is a good idea...

**Comment by benoitdm-oslandia** [comment]
Ok so no need to go further and we will update the documentation accordingly.

**Comment by azarz** [comment]
We should also document that for POST, PUT and PATCH, the crs is defined by the `Content-Crs` header. I don't know where to specify that in the openApi definition.

**Comment by azarz** [comment]
> We should also document that for POST, PUT and PATCH, the crs is defined by the `Content-Crs` header. I don't know where to specify that in the openApi definition.

Done

**Review by benoitdm-oslandia — DISMISSED** [review]
LGTM

**Review by jmkerloch — DISMISSED** [review]
LGTM

**Comment by azarz** [comment]
New review required since I rebased on develop @jmkerloch @benoitdm-oslandia (I didn't change anything)

**Review by jmkerloch — APPROVED** [review]

**Pull request merged**